pkg/handler3/handler_test.go

```diff
@@ -432,3 +434,118 @@ func TestCacheBusting(t *testing.T) {
 		}
 	}
 }
+
+func MarshalOrDie(spec *spec3.OpenAPI) []byte {
+	j, err := json.Marshal(spec)
+	if err != nil {
+		panic(err)
+	}
+	return j
+
+}
+
+func OpenAPIOrDie(name string) *spec3.OpenAPI {
+	openapi := fmt.Sprintf(`{
+  "openapi": "3.0",
+  "info": {
+   "title": "%s",
+   "version": "v1.23.0"
+  },
+  "paths": {}}`, name)
+	spec := spec3.OpenAPI{}
+	if err := json.Unmarshal([]byte(openapi), &spec); err != nil {
+		panic(err)
+	}
+	return &spec
+}
+
+func getDiscovery(server *httptest.Server, path string) (*OpenAPIV3Discovery, string, error) {
+	client := server.Client()
+	req, err := http.NewRequest("GET", server.URL+"/"+path, nil)
+	if err != nil {
+		return nil, "", fmt.Errorf("error in creating new request: %v", err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, "", fmt.Errorf("error in serving HTTP request: %v", err)
+	}
+	if resp.StatusCode != 200 {
+		return nil, "", fmt.Errorf("unexpected response status code, want: %v, got: %v", 200, resp.StatusCode)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, "", fmt.Errorf("Failed to read request body: %v", err)
+	}
+
+	discovery := &OpenAPIV3Discovery{}
+	if err := json.Unmarshal(body, &discovery); err != nil {
+		return nil, "", fmt.Errorf("failed to unmarshal discovery: %v", err)
+	}
+	return discovery, resp.Header.Get("etag"), nil
+}
+
+func TestUpdateSpecLazy(t *testing.T) {
+	mux := http.NewServeMux()
+	o := NewOpenAPIService()
+
+	mux.Handle("/openapi/v3", http.HandlerFunc(o.HandleDiscovery))
+	mux.Handle("/openapi/v3/apis/apps/v1", http.HandlerFunc(o.HandleGroupVersion))
```

Comment:
I don't see anywhere this path is being requested. I'd assume either we'd want to check the contents of the path (as opposed to just discovery), or this line can probably be removed since the contents of the path is not of interest in the current test.

Reply:
Yeah, I mostly wanted to see how the discovery is behaving. I'm assuming this endpoint is already tested by the test above.

Reply:
I've taken the line out.